certification/internal/authn/keychain.go

```diff
@@ -0,0 +1,81 @@
+package authn
+
+import (
+	"os"
+	"sync"
+
+	"github.com/docker/cli/cli/config"
+	"github.com/docker/cli/cli/config/types"
+	craneauthn "github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+)
+
+type preflightKeychain struct {
+	mu sync.Mutex
```

Comment:
Pardon my ignorance, but why do we need mutual exclusion here?

The only reason I ask is because this feels like a _lot_ of code for what is otherwise just "read creds or assume anonymous". 

We don't run checks concurrently, but even if we did, I'm of the opinion that the credentials in use should _never_ change within the context of check execution, and should only ever change at the outset of execution (e.g. initializing everything before we start our core logic).

Reply:
Complete honesty: This was cargo-culted from the DefaultKeychain in go-containerregistry.

Reply:
Removed.